include/bm/bm_sim/data.h

```diff
@@ -178,30 +178,27 @@ class Data {
   //! Convert the value of Data to any inegral type
   template<typename T,
            typename std::enable_if<std::is_integral<T>::value, int>::type = 0>
-  T get() const {
+  typename std::remove_const<T>::type get() const {
```

Comment:
does the `const` need to be removed here as well? That could be surprising for the client, I think it should stay in the return type.